tests/FSharp.Compiler.ComponentTests/CompilerOptions/fsc/warnon/warnon.fs

```diff
@@ -34,3 +34,83 @@ module warnon =
         |> withDiagnosticMessageMatches "The value 'n' is unused$"
         |> ignore
 
+    [<Fact>]
```

Comment:
This might not be the best place to put these tests, this file has some autogenerated tests for fsc options. Don't know, off the top of my head, where it should go.

Reply:
I would say it is fine to have those tests here too, as long as it fits the category